src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -253,5 +243,11 @@ internal static async Task<string> GetRoutePrefix(string rootScriptPath)
 
             return routePrefix;
         }
+
+        internal static string GetBaseUrl()
+        {
+            string hostName = Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME") ?? "localhost";
```

Comment:
Previously we were getting the base URL from the active request. However, we're now needing to compute this in the absence of a request (e.g. background sync triggers)

Reply:
Should this be using `IEnvironment` rather than `Environment`?

Reply:
I'll be rebasing on the WEBSITE_HOSTNAME changes that just went in so this will be coming from the HostNameProvider